lib/node_modules/@stdlib/math/base/special/besselj1/benchmark/r/benchmark.R

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env Rscript
+
+# Set the precision to 16 digits:
+options( digits = 16 );
+
+#' Run benchmarks.
+#'
+#' @examples
+#' main();
+main <- function() {
+	# Define benchmark parameters:
+	name <- "besselj1";
+	iterations <- 1000000L;
+	repeats <- 3;
+
+	#' Print the TAP version.
+	#'
+	#' @examples
+	#' print_version();
+	print_version <- function() {
+		cat( "TAP version 13\n" );
+	}
+
+	#' Print the TAP summary.
+	#'
+	#' @param total Total number of tests.
+	#' @param passing Total number of passing tests.
+	#'
+	#' @examples
+	#' print_summary( 3, 3 );
+	print_summary <- function( total, passing ) {
+		cat( "#\n" );
+	    cat( paste0( "1..", total, "\n" ) ); # TAP plan
+	    cat( paste0( "# total ", total, "\n" ) );
+	    cat( paste0( "# pass  ", passing, "\n" ) );
+	    cat( "#\n" );
+	    cat( "# ok\n" );
+	}
+
+	#' Print benchmark results.
+	#'
+	#' @param iterations Number of iterations.
+	#' @param elapsed Elapsed time in seconds.
+	#'
+	#' @examples
+	#' print_results( 10000L, 0.131009101868 );
+	print_results <- function( iterations, elapsed ) {
+		rate <- iterations / elapsed;
+	    cat( "  ---\n" );
+	    cat( paste0( "  iterations: ", iterations, "\n" ) );
+	    cat( paste0( "  elapsed: ", elapsed, "\n" ) );
+	    cat( paste0( "  rate: ", rate, "\n" ) );
+	    cat( "  ...\n" );
+	}
+
+	#' Run a benchmark.
+	#'
+	#' ## Notes
+	#'
+	#' * We compute and return a total "elapsed" time, rather than the minimum
+	#'   evaluation time, to match benchmark results in other languages (e.g.,
+	#'   Python).
+	#'
+	#'
+	#' @param iterations Number of Iterations.
+	#' @return Elapsed time in seconds.
+	#'
+	#' @examples
+	#' elapsed <- benchmark( 10000L );
+	benchmark <- function( iterations ) {
+		# Run the benchmarks:
+		results <- microbenchmark::microbenchmark( besselJ( runif(1)*100.0, 1 ), times = iterations );
```

Comment:
Value regime should match that of `C` and other benchmarks. The reason, conceptually, is to have the primary "independent variable" be language and prevent some benchmarks from overly representing certain value ranges over others which may put that language at an inherent disadvantage. By keeping test values the same, we hopefully treat all langs more or less the same.